src/fparser/Fortran2003.py

```diff
@@ -552,16 +562,29 @@ class BracketBase(Base):
     <bracket-base> = <left-bracket-base> <something> <right-bracket>
     """
     def match(brackets, cls, string, require_cls=True):
-        i = len(brackets)/2
-        left = brackets[:i]
-        right = brackets[-i:]
+        bracket_len = len(brackets)/2
+        left = brackets[:bracket_len]
+        right = brackets[-bracket_len:]
         if string.startswith(left) and string.endswith(right):
-            line = string[i:-i].strip()
+            # We may have something like "(a + b)*(a - b)" so have
```

Comment:
Sadly you were right. In fact some of my indexing into the string wasn't correct either. I've extended the logic now to ignore anything that occurs between '..' or "..." chars. I've also extended the associated test. cd33651